### PR TITLE
[Bug Fix] Add new `BTShopperInsightsError.invalidAuthorization`

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Add new `BTShopperInsightsError.invalidAuthorization` and check authorization type in `BTShopperInsightsClient`
    - This fixes a bug where you would see `SWIFT TASK CONTINUATION MISUSE: post(_:parameters:headers:httpType:) leaked its continuation!` when using shopper insights with a tokenization key

### Checklist

- [x] Added a changelog entry

### Authors

- @jaxdesmarais

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

File: UnitTests/BraintreeShopperInsightsTests/BTShopperInsightsClient_Tests.swift [hunk 1]
```diff
@@ -6,7 +6,8 @@ import XCTest
 
 class BTShopperInsightsClient_Tests: XCTestCase {
     
-    var mockAPIClient = MockAPIClient(authorization: "development_client_key")!
+    let clientToken = TestClientTokenFactory.token(withVersion: 3)
+    var mockAPIClient: MockAPIClient?
```
Comment (scannillo) [inline UnitTests/BraintreeShopperInsightsTests/BTShopperInsightsClient_Tests.swift:10]: We do this a lot in our tests so we don't have to unwrap the optional in each test case (similar to what we do for the `sut`)

```suggestion
    var mockAPIClient: MockAPIClient!
```

File: Sources/BraintreeShopperInsights/BTShopperInsightsClient.swift [hunk 2]
```diff
@@ -39,7 +39,11 @@ public class BTShopperInsightsClient {
             email: request.email,
             phone: request.phone
         )
-        
+
+        if apiClient.authorization.type != .clientToken {
```
Comment (scannillo) [inline Sources/BraintreeShopperInsights/BTShopperInsightsClient.swift:43]: Take it or leave it for readability - move this above the `postParameters` construction

**Review by scannillo — APPROVED** [review]
A few small comments, but otherwise looks good!

Do we have a ticket to tackle this on Android?

**Review by richherrera — APPROVED** [review]

File: Sources/BraintreeShopperInsights/BTShopperInsightsClient.swift [hunk 3]
```diff
@@ -39,7 +39,11 @@ public class BTShopperInsightsClient {
...
+            throw notifyFailure(with: BTShopperInsightsError.invalidAuthorization)
```
Comment (KunJeongPark) [inline Sources/BraintreeShopperInsights/BTShopperInsightsClient.swift:44]: Nice!

**Review by KunJeongPark — APPROVED** [review]

**Pull request merged**